### ISSUE TITLE
NullPointerException on getting gitlab projects

### ISSUE THREAD

**Issue by vpapako** [hobbit-project/platform#362]
PlatformController fails to check if a gitlab user is an admin with a `NullPointerException`.
It seems that `GitlabUser` successfully getted (e.g. I am getting my username if request it), but the `isAdmin()` method returns null.

Tested on both hobbitproject/hobbit-platform-controller:latest (from Docker Hub) and locally built images

```
2018-10-31 08:31:47,568 ERROR [org.hobbit.controller.gitlab.GitlabControllerImpl] - <Couldn't get all gitlab projects.>
java.lang.NullPointerException
	at org.hobbit.controller.gitlab.GitlabControllerImpl$2.run(GitlabControllerImpl.java:148)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)
```

https://github.com/hobbit-project/platform/blob/fc87fbaa4d4e96ac8ac68e10ed0252bce8100e47/platform-controller/src/main/java/org/hobbit/controller/gitlab/GitlabControllerImpl.java#L148

**Comment by vpapako** [comment]
Some more info:

If I completely remove the check, so the following line will be executed

https://github.com/hobbit-project/platform/blob/fc87fbaa4d4e96ac8ac68e10ed0252bce8100e47/platform-controller/src/main/java/org/hobbit/controller/gitlab/GitlabControllerImpl.java#L151

it works fine and gets all projects (Projects: 226).

On the other hand, if I let the else part to be executed

https://github.com/hobbit-project/platform/blob/fc87fbaa4d4e96ac8ac68e10ed0252bce8100e47/platform-controller/src/main/java/org/hobbit/controller/gitlab/GitlabControllerImpl.java#L155

the following exception is thrown

```
java.lang.RuntimeException: java.io.FileNotFoundException: https://git.project-hobbit.eu/api/v4/projects/visible
 	at org.gitlab.api.http.GitlabHTTPRequestor$1.fetch(GitlabHTTPRequestor.java:243)
 	at org.gitlab.api.http.GitlabHTTPRequestor$1.hasNext(GitlabHTTPRequestor.java:197)
 	at org.gitlab.api.http.GitlabHTTPRequestor.getAll(GitlabHTTPRequestor.java:165)
 	at org.hobbit.controller.gitlab.GitlabControllerImpl$2.run(GitlabControllerImpl.java:155)
 	at java.util.TimerThread.mainLoop(Timer.java:555)
 	at java.util.TimerThread.run(Timer.java:505)
 Caused by: java.io.FileNotFoundException: https://git.project-hobbit.eu/api/v4/projects/visible
 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1890)
 	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
 	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
 	at org.gitlab.api.http.GitlabHTTPRequestor.parse(GitlabHTTPRequestor.java:363)
 	at org.gitlab.api.http.GitlabHTTPRequestor.access$200(GitlabHTTPRequestor.java:38)
 	at org.gitlab.api.http.GitlabHTTPRequestor$1.fetch(GitlabHTTPRequestor.java:236)
 	... 5 more

```

**Comment by denkv** [comment]
Both problems should be now fixed in `hotfix/gitlab-api-v4` branch. PR: #363

**Comment by vpapako** [comment]
I think that the `getProjectsOfUser` method needs to be updated accordingly.

**Comment by denkv** [comment]
Fixed in v2.0.7.
